apps/yggdrasil/test/yggdrasil/user_role.exs

```diff
@@ -0,0 +1,104 @@
+defmodule Yggdrasil.UserRoleTest do
+  use ExUnit.Case, async: false
+
+  alias Yggdrasil.{Repo, User, Role, UserRole, RoleResource, Resource, Permission}
+
+  @user %{username: "tester",
+          password: "password",
+          password_confirmation: "password"}
+
+  @test_perms [:foo_perm, :bar_perm, :baz_perm]
+  @test_resources [:foo_res, :bar_res, :baz_res]
+  @test_roles [
```

Comment:
This is begging to be extract out into a module, I have been lugging around you will see similar code in the ensure_permission and user tests.

I don't know exactly what the best practice here, obviously copying the code around is not one of them, but possible just need some central role and user helper module for fixtures in the test. Similar to how phoenix provides the model, conn, and channel cases.